src/Utility/Hash.php

```diff
@@ -370,7 +365,9 @@ protected static function _simpleOp($op, $data, $path, $values = null)
                 }
             } elseif ($op === 'remove') {
                 if ($i === $last) {
-                    unset($_list[$key]);
+                    if (!is_string($_list)) {
```

Comment:
Hmm, but taking the performance into consideration, doing the check at the current place may be better. I'll change `!is_string()` to `is_array()`.